### ISSUE TITLE
Chunk conversion op silently produces unequal chunks when the fused tensor is not divisible by the target count

### ISSUE THREAD

**Issue by VaggelisGian** [huggingface/transformers#48323]
### System Info

- transformers 5.16.0.dev0 (current main)
- python 3.12.6, Windows 11 (not platform specific)

### Reproduction

The v5 dynamic loading API's `Chunk` op (`src/transformers/core_model_loading.py`, class `Chunk`) splits a fused tensor for targets like q/k/v. Its guard only compares chunk COUNT against the target count:

```python
chunks = tuple(chunk.contiguous() for chunk in torch.chunk(tensor, sizes, dim=self.dim))
if len(input_dict) > 1 or len(target_patterns) == 1 or len(chunks) != len(target_patterns):
    raise ValueError(f"Failed to convert {kwargs.get('full_layer_name')}")
```

`torch.chunk` returns unequal chunks when the tensor size is not divisible by the requested count, and can still return exactly as many chunks as there are targets. Then every downstream consumer receives differently sized tensors:

```python
import torch
from transformers.core_model_loading import Chunk

op = Chunk(dim=0)
t = {"x": [torch.arange(5.)]}
out = op.convert(t, source_patterns=["x"], target_patterns=["a", "b"], full_layer_name="demo.layer")
print({k: tuple(v.shape) for k, v in out.items()})
```

Output on current main - no error, wrong shapes:

```text
{'a': (3,), 'b': (2,)}
```

In a real `from_pretrained` conversion this surfaces either as a confusing per-target shape mismatch error far from the cause, or worse: silently re-initializing weights under `ignore_mismatched_sizes`, which corrupts the model without any signal.

### Expected behavior

Since the op's own contract is "Split a tensor along dim into equally sized chunks", an indivisible split should fail loudly at the conversion site with a clear message naming the violated contract and the layer, instead of producing silently unequal chunks. Every registered `Chunk` usage in the codebase splits equal-sized siblings by construction, so a divisibility check would not reject any legitimate existing use.

Duplicate check: searched open/closed issues and PRs around torch.chunk uneven splits, Chunk.convert, and ignore_mismatched_sizes silent re-init; nothing tracks this.

**Comment by Ki-Seki** [comment]
Hi @Cyrilvallez, I’d be interested in adding the divisibility validation and a focused regression test for Chunk.convert. Would it be okay for me to take it on?

**Comment by Cyrilvallez** [comment]
Theoretical issue never true in practice
